content/posts/cross-compile-query-json.mdx

```diff
@@ -0,0 +1,252 @@
+---
+title: "query-json: A story of cross-compilation with Reason"
+date: "2020-10-12"
+tags:
+  - "query-json"
+  - "json"
+  - "jq"
+  - "Reason"
+  - "OCaml"
+  - "js_of_ocaml"
+  - "jsoo"
+---
+
+<!-- TODO: Disclamer Difference between Rescript/Reason/BuckleScript/OCaml -->
+# What's query-json
+
+[query-json](https://github.com/davesnx/query-json) is a faster and simpler re-implementation of [jq's language](https://github.com/stedolan/jq/wiki/jq-Language-Description) in Reason Native.
+
+It's a CLI to run small programs against JSON, the same idea as [sed](https://www.linode.com/docs/tools-reference/tools/manipulate-text-from-the-command-line-with-sed/), but for JSON files. As a web engineer is an essential tool while developing APIs.
+
+Started with the goal to create something useful and learn during the process. Was very interested in learning about how to write parsers and compilers using the OCaml stack: [menhir](https://opam.ocaml.org/packages/menhir) and [sedlex](https://github.com/ocaml-community/sedlex), compile it to any architecture and try to compile it to JavaScript.
+
+This post explains the project, how it was made and which decisions where followed and some reflections.
+
+## Why learning parsers/compilers
+I had a little idea about the theory, but very vague and useless. Since I create [styled-ppx](https://github.com/davesnx/styled-ppx) which is a
+ppx (**P**re**P**rocessor E**x**tension) that allows CSS-in-Reason/OCaml. Needs to parse the CSS and have a backend that compiles to [bs-emotion](https://github.com/ahrefs/bs-emotion).
+
+I asked for some help to [@EduardoRFS](https://www.twitch.tv/eduardorfs) about writting a CSS Parser that supports the entire [CSS3 specification](https://www.w3.org/TR/2001/WD-css3-roadmap-20010523/) which I will want to understand, improve and maintain over time.
+
+## How it works
+```bash
+query-json ".store.books | filter(.price > 10)" stores.json
+```
+
+This would read `stores.json` and access to `"store"` field, access to `"books"` field, since it's an array, it will run a filter on each item and if `"price" `field is bigger than 10 will keep that item on the list and finally print the resultant list.
```

Comment:
```suggestion
This would read `stores.json` and access to `"store"` field, access to `"books"` field, since it's an array, it will run a filter on each item and if `"price" `field is larger than 10, it will keep that item on the list and finally print the resultant list.
```